dot/state/pruner.go

```diff
@@ -0,0 +1,363 @@
+package state
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ChainSafe/chaindb"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/scale"
+)
+
+const (
+	journalPrefix = "journal"
+	lastPrunedKey = "last_pruned"
+	pruneInterval = time.Second
+)
+
+// nolint
+const (
+	FullNode    = "full"
```

Comment:
any reason why these aren't of type `Pruning`?

Reply:
Done